_posts/2021-05-10-helm-charts-for-wildfly.adoc

```diff
@@ -0,0 +1,225 @@
+----
+
+The application is up and running.
+Let's now look at the configuration file that the Chart used to install the application.
+
+## Looking at the Helm Release File
+
+Let's deep dive in the https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml[YAML File] that we use to install the application (https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml):
+
+[source,yaml]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  ref: 23.0.2.SP1
+  mode: bootable-jar
+  env:
+  - name: ARTIFACT_DIR
+    value: microprofile-config/target
+  - name: MAVEN_ARGS_APPEND
+    # Use the bootable-jar-openshift profile to ensure that the application
+    # can be deployed on OpenShift but disable JKube as the image will be 
+    # built and deployed by this chart.
+    value: -am -pl microprofile-config -Pbootable-jar-openshift -Djkube.skip=true
+  - name: MAVEN_OPTS
+    value: '-XX:MetaspaceSize=251m -XX:MaxMetaspaceSize=256m'
+deploy:
+  replicas: 1
+  env:
+  - name: CONFIG_PROP
+    value: Hello from OpenShift
+----
+
+The first thing to notice is that the configuration file contains two main sections `build` and `deploy`.
+
+The `build` section focuses on building the application image using an OpenShift `BuildConfig` resource.
+The `BuildConfig` resource pulls the application from a Git repository (based on the `uri` and `ref` fields).
+
+The `mode` field specifies which types of application image to build. There are two valid modes: `s2i` and `bootable-jar`.
+
+The `s2i` build mode uses WildFly Source-to-Image (S2I) Builder and Runtime images to create the application image.  
+In the `bootable-jar` build mode, the `BuildConfig` will compile the application as a Bootable Jar and use the OpenJDK 11 image as the base image.
+
+Finally there is an `env` section that contains any environment variables needed to *build* the image. We have three environment variables that are needed for Maven options.
+As a whole they mean that we build only the specific `microprofile-config` Maven module from the quickstart Git repository with the `bootable-jar-openshift` Maven Profile (to create a Bootable Jar) and use its artifact as the target of the application image.
+
+The `deploy` section focuses on deploying the application image on OpenShift. It creates different resources (`Deployment`, one or many `Services`, a `Route`) to make the application accessible from inside and outside the cluster by default.
+
+We have only defined two fields in that section:
+
+* `replicas` which specifies the number of pods that the application will use
+* `env` wich are environment variables needed to *run* the image. In our example, we only have one named `CONFIG_PROP` and its value is used in the HTTP endpoint's response.
+
+There are a lot of things to discuss with this chart and we will have other blog post to showcase its features.
```

Comment:
@jmesnil 

Either "another blog post" or "other blog posts"  -- which depends on whether you plan more than one. :)

Reply:
I left a plural so I committed to write at least 2 more :) 

Actually, I think there will be 3 follow ups:
* one about the OpenShift Dev console UX
* one about more advance use cases (based on the `todo-backend` quickstart)
* one about adapting the Chart to deploy on vanilla Kubernetes clusters